BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -0,0 +1,400 @@
+                else { return }
+            self.value = dateVal
+            
+        case "HKBiologicalSex":
+            guard let val = value as? HKBiologicalSex else { return }
+            self.value = val
+            
+        case "HKQuantity":
+            guard let val = value as? NSNumber else { return }
+            self.value = val
+            
+        default:
+            break
+        }
+    }
+    
+    func commonDemographicJsonValue() -> SBBJSONValue? {
+        guard let jsonVal = self.commonJsonValueGetter() else { return nil }
+        if self.itemType == "HKBiologicalSex" {
+            return (self.value as? HKBiologicalSex)?.demographicDataValue
+        }
+        
+        return jsonVal
+    }
+    
+    func commonCheckTypeCompatible(newValue: Any?) -> Bool {
+        guard newValue != nil else { return true }
+        
+        switch self.itemType {
+        case "String":
+            return newValue as? String != nil
+            
+        case "Number":
+            return newValue as? NSNumber != nil
+            
+        case "Bool":
+            return newValue as? NSNumber != nil
+            
+        case "Date":
+            return newValue as? NSDate  != nil
+            
+        case "HKBiologicalSex":
+            return newValue as? HKBiologicalSex != nil
+            
+        case "HKQuantity":
+            return newValue as? NSNumber != nil
+            
+        default:
+            return false
+        }
+    }
+}
+
+class SBAProfileItemBase: NSObject, SBAProfileItem {
+    /**
+     The value property is used to get and set the profile item's value in whatever internal data
+     storage is used by the implementing class.
+     */
+    open var value: Any?
+
+    fileprivate let sourceDict: [AnyHashable: Any]
+    
+    open var key: String {
+        get {
+            return sourceDict["key"]! as! String
+        }
+    }
+    
+    open var sourceKey: String {
+        get {
+            return sourceDict["sourceKey"] as! String? ?? self.key
+        }
+    }
+    
+    open var demographicKey: String {
+        get {
+            return sourceDict["demographicKey"] as! String? ?? self.key
+        }
+    }
+    
+    open var itemType: String {
+        get {
+            return sourceDict["itemType"]! as! String
+        }
+    }
+    
+    public required init(dictionaryRepresentation dictionary: [AnyHashable: Any]) {
+        sourceDict = dictionary
+        super.init()
+    }
+    
+    open var jsonValue: SBBJSONValue? {
+        get {
+            return self.commonJsonValueGetter()
+        }
+        
+        set {
+            commonJsonValueSetter(value: newValue)
+        }
+    }
+    
+    open var demographicJsonValue: SBBJSONValue? {
+        get {
+            return self.commonDemographicJsonValue()
+        }
+    }
+}
+
+class SBAKeychainProfileItem: SBAProfileItemBase {
+    private var keychain: SBAKeychainWrapper
```

Comment:
because it would only let me set it once in the initializer--fixed by using a temp var until I'm sure what to set it to